pkg/machines/scripts/install_machine.sh

```diff
@@ -47,7 +48,9 @@ else
     GRAPHICS_PARAM="$CREATE_OPTIONS_RESULT"
 fi
 
-if [ "${SOURCE#/}" != "$SOURCE" ] && [ -f "${SOURCE}" ]; then
+if [ "$SOURCE_TYPE" == "pxe" ]; then
+        LOCATION_PARAM="--pxe --network $SOURCE"
```

Comment:
Before this change we don't specify `--network` when executing `virt-install` command which means that the created/installed guest will have the `default` NAT network.  With this change when the guest is installed using `PXE` method it will not have the `default` network like for all other installation methods but the selected network in the creation dialog.

The reason why I'm mentioning it is that for filesystem or URL installation source type the installation source data are used only for installation and are removed after the installation is done, but in case of PXE the source is not used only for installation but it configures the main and only network for the guest once the installation is done.

The question is do we want to have it the same way as virt-manager that in the creation dialog you need to select the installation source or will we leave the guest interface configuration up to the user?  In both cases we should improve guest interface configuration because right now you can only edit existing interface but you cannot add/remove interfaces.

Reply:
Good point, thanks for raising these issues/thoughts. So, I think we should definitely allow the iser to select the installation source, whatever that is. Otherwise the user will never be able to start immediately a VM from PXE if his default network is not configured like that. Indeed the NICs dialog needs enhancements as well. But I don't think we should rely on that the user will create the VM with PXE install mode and then go to the Interfaces tab to configure them before starting the VM.
And of course we need to find a more decent to offer to users who need advanced configurations, since this applies to other parameters apart from pxe booting.